lib_standard_app/debug.h

```diff
@@ -1,3 +1,168 @@
 #pragma once
+#include "macros.h"
+#include "os.h"
 
-WEAK void __attribute__((noreturn)) debug_display_throw_error(int exception);
+#ifdef HAVE_DEBUG_THROWS
+WEAK void app_throw_info(unsigned int exception, unsigned int lr_val);
+WEAK void __attribute__((noreturn)) debug_display_exit(int exception);
+#endif
+
+void ledger_assert(void);
+
+#define ASSERT_ERROR_CODE 0xBAAD
+
+#ifdef LEDGER_ASSERT_CONFIG_FILE_INFO
+#define LEDGER_ASSERT_CONFIG_MESSAGE_INFO   1
+#define LEDGER_ASSERT_CONFIG_LR_AND_PC_INFO 1
+#endif
+
+#ifdef LEDGER_ASSERT_CONFIG_MESSAGE_INFO
+#define LEDGER_ASSERT_CONFIG_LR_AND_PC_INFO 1
+#endif
+
+#ifdef LEDGER_ASSERT_CONFIG_LR_AND_PC_INFO
+#endif
+
+#ifdef HAVE_LEDGER_ASSERT_DISPLAY
+#ifdef LEDGER_ASSERT_CONFIG_LR_AND_PC_INFO
+#define LR_AND_PC_OFFSET 0
+#define LR_AND_PC_SIZE   30
+WEAK void assert_display_lr_and_pc(int lr, int pc);
+#define ASSERT_DISPLAY_LR_AND_PC(lr, pc) assert_display_lr_and_pc(lr, pc)
+#else
+#define LR_AND_PC_SIZE 0
+#endif
```

Comment:
I think we could inverse the `#ifdef` order to make it easier to read.
It would be something like:

```
#if defined(LEDGER_ASSERT_CONFIG_LR_AND_PC_INFO) && defined(HAVE_LEDGER_ASSERT_DISPLAY)
#define LR_AND_PC_OFFSET 0
#define LR_AND_PC_SIZE   30
WEAK void assert_display_lr_and_pc(int lr, int pc);
#define ASSERT_DISPLAY_LR_AND_PC(lr, pc) assert_display_lr_and_pc(lr, pc)
#else
#define LR_AND_PC_SIZE 0
#define ASSERT_DISPLAY_LR_AND_PC(lr, pc) \
    do {                                 \
    } while (0)
#endif

#if defined(LEDGER_ASSERT_CONFIG_LR_AND_PC_INFO) && defined(HAVE_PRINTF)
void print_lr_and_pc(int lr, int pc);
#define ASSERT_PRINT_LR_AND_PC(lr, pc) print_lr_and_pc(lr, pc)
#else
#define ASSERT_PRINT_LR_AND_PC(lr, pc) \
    do {                               \
    } while (0)
#endif
```
how does that sounds to you?

Reply:
Good idea ! I had a nice formatting but I guess the linter got rid of it :smiling_face_with_tear: